tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -816,32 +816,21 @@ def program_domain_tuple(
 
 
 def test_where_k_offset(cartesian_case):
-    if cartesian_case.backend in [gtfn_cpu.run_gtfn, gtfn_cpu.run_gtfn_imperative]:
-        pytest.xfail("IndexFields are not supported yet.")
-
     @gtx.field_operator
     def fieldop_where_k_offset(
-        a: cases.IKField, k_index: gtx.Field[[KDim], gtx.IndexType]
+        inp: cases.IKField, k_index: gtx.Field[[KDim], gtx.IndexType]
     ) -> cases.IKField:
-        return where(k_index > 0, a(Koff[-1]), 2)
+        return where(k_index > 0, inp(Koff[-1]), 2)
 
-    cases.verify_with_default_data(
-        cartesian_case,
-        fieldop_where_k_offset,
-        ref=lambda a, k_index: np.where(k_index > 0, np.roll(a, 1, axis=1), 2),
-    )
+    inp = cases.allocate(cartesian_case, fieldop_where_k_offset, "inp")()
+    k_index = cases.allocate(
+        cartesian_case, fieldop_where_k_offset, "k_index", strategy=cases.IndexInitializer()
+    )()
+    out = cases.allocate(cartesian_case, fieldop_where_k_offset, "inp")()
 
-    @gtx.field_operator
-    def fieldop_where_k_offset(
-        a: cases.IKField, k_index: gtx.Field[[KDim], gtx.IndexType]
-    ) -> cases.IKField:
-        return where(k_index > 0, a(Koff[-1]), 2)
+    ref = np.where(np.asarray(k_index) > 0, np.roll(inp, 1, axis=1), 2)
 
-    cases.verify_with_default_data(
-        cartesian_case,
-        fieldop_where_k_offset,
-        ref=lambda a, k_index: np.where(k_index > 0, np.roll(a, 1, axis=1), 2),
-    )
+    cases.verify(cartesian_case, fieldop_where_k_offset, inp, k_index, out=out, ref=ref)
```

Comment:
is this the best I can do with the test helper? Note I introduce a new IndexInitializer